### ISSUE TITLE
Xcode compile error in JSONEncoedable

### ISSUE THREAD

**Issue by simondh** [bignerdranch/Freddy#172]
Snippet:
https://bitbucket.org/snippets/simon_hewitt/aKjng

First attempt to use JSONEncodeable, I get 
"Type of expression is ambiguous without more context"
at line 270 in the snippet. I have tried  adding [String:Any] but cannot get it right - Any help please!

**Comment by jgallagher** [comment]
The error you're getting is very unhelpful, but I believe the problem is that you have the force-unwrap in the wrong place. Try moving the `!` inside the parens:

```swift
extension PermissionData: JSONEncodable {
    public func toJSON() -> JSON {
        return .Dictionary ([
            "permission_id" : .Int(permission_id!),
            "permission_name": .String(permission_name!),
            "permission" : .Bool(permission!)
            ])
    }
}
```

although do note that this will crash if any of those properties are `nil`. You may want to either make the properties non-optional or provide default values inside `toJSON()` if they are `nil`.

**Comment by simondh** [comment]
Yes that was it! Many thanks
As you say the error does not help much, so thanks for helping me.

On 30 May 2016 at 17:15, John Gallagher <notifications@github.com> wrote:

> Closed #172 <https://github.com/bignerdranch/Freddy/issues/172>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bignerdranch/Freddy/issues/172#event-676195043>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AIFv38pQrCZjA1xChWDX-27-uEmeMl7aks5qGw09gaJpZM4Ip5Fu>
> .
>



-- 
Simon Hewitt
(+44) 07584 658227
